portfolio/portfolio.go

```diff
@@ -472,70 +507,81 @@ func (b *Base) GetPortfolioGroupedCoin() map[currency.Code][]string {
 	return result
 }
 
-// Seed appends a portfolio base object with another base portfolio
-// addresses
-func (b *Base) Seed(port Base) {
-	b.Addresses = port.Addresses
-}
-
 // StartPortfolioWatcher observes the portfolio object
-func (b *Base) StartPortfolioWatcher() {
-	addrCount := len(b.Addresses)
-	log.Debugf(log.PortfolioMgr,
-		"PortfolioWatcher started: Have %d entries in portfolio.\n", addrCount,
-	)
+func (b *Base) StartPortfolioWatcher(ctx context.Context, sleepDelay time.Duration) error {
+	if len(b.Addresses) == 0 {
+		return errNoPortfolioItemsToWatch
+	}
+
+	if sleepDelay <= 0 {
+		sleepDelay = time.Minute * 10
+	}
+
+	log.Infof(log.PortfolioMgr, "PortfolioWatcher started: Have %d entries in portfolio.\n", len(b.Addresses))
+
+	ticker := time.NewTicker(sleepDelay)
+	defer ticker.Stop()
+
```

Comment:
🚧 Do we really want to wait Interval before first fetch?
That is: If the user has set it to 1 hour, justifiably, should they have to wait before getting any data populated?

🥃 Kinda never tried this, but I wonder if we could stuff something down the channel to cause a premature ejaculation?
```suggestion
ticker.C <- time.Now()
```

Reply:
Good point, I've now fired it immediately before starting the ticker